packages/generator/src/edmx-to-vdm/v4/function-import.ts

```diff
@@ -10,21 +11,65 @@ import {
 import { ServiceMetadata } from '../../edmx-parser/edmx-file-reader';
 import { VdmComplexType, VdmEntity, VdmFunctionImport } from '../../vdm-types';
 import { parseFunctionImportReturnTypes } from '../common/action-function-return-types';
+import { hasUnsupportedParameterTypes } from '../edmx-to-vdm-util';
 import { findActionFunctionByImportName } from './action-function-util';
 
+const logger = createLogger({
+  package: 'generator',
+  messageContext: 'function-import'
+});
+
 function findFunctionForFunctionImport(
   functions: EdmxFunction[],
   functionImport: EdmxFunctionImport
-): EdmxFunction {
-  return findActionFunctionByImportName(
-    functions,
-    functionImport.Function,
-    'function'
-  );
+): EdmxFunction | undefined {
+  return findActionFunctionByImportName(functions, functionImport.Function);
 }
 
 const extractResponse = (response: string) => `${response}.value`;
 
+interface JoinedFunctionImportData {
+  functionImport: EdmxFunctionImport;
+  function: EdmxFunction;
+}
+
+function joinFunctionImportData(
+  functionImports: EdmxFunctionImport[],
+  functions: EdmxFunction[]
+): JoinedFunctionImportData[] {
+  const functionImportsWithoutFunctions: string[] = [];
```

Comment:
```suggestion
  const functionImportsWithoutFunctions = [];
```

Reply:
it won't work due to `Argument of type 'string' is not assignable to parameter of type 'never'.`

Reply:
Then probably `: EdmxFunctionImport[]` with my suggestions below.